_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1024 @@
+
+```json
+GET /_plugins/_ml/tasks/aFeif4oB5Vm0Tdw8yoN7
+```
+{% include copy-curl.html %}
+
+Once the task is complete, the task state will be `COMPLETED` and the Tasks API response will contain a model ID for the registered model:
+
+```json
+{
+  "model_id": "aVeif4oB5Vm0Tdw8zYO2",
+  "task_type": "REGISTER_MODEL",
+  "function_name": "TEXT_EMBEDDING",
+  "state": "COMPLETED",
+  "worker_node": [
+    "4p6FVOmJRtu3wehDD74hzQ"
+  ],
+  "create_time": 1694358489722,
+  "last_update_time": 1694358499139,
+  "is_async": true
+}
+```
+
+You'll need the model ID in order to use this model for several of the following steps.
+
+<details closed markdown="block">
+  <summary>
+    Test it
+  </summary>
+  {: .text-delta}
+
+Search for the newly created model by providing its ID in the request:
+
+```json
+GET /_plugins/_ml/models/aVeif4oB5Vm0Tdw8zYO2
+```
+{% include copy-curl.html %}
+
+The response contains the model:
+
+```json
+{
+  "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
+  "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+  "algorithm": "TEXT_EMBEDDING",
+  "model_version": "1",
+  "model_format": "TORCH_SCRIPT",
+  "model_state": "REGISTERED",
+  "model_content_size_in_bytes": 266352827,
+  "model_content_hash_value": "acdc81b652b83121f914c5912ae27c0fca8fabf270e6f191ace6979a19830413",
+  "model_config": {
+    "model_type": "distilbert",
+    "embedding_dimension": 768,
+    "framework_type": "SENTENCE_TRANSFORMERS",
+    "all_config": """{"_name_or_path":"old_models/msmarco-distilbert-base-tas-b/0_Transformer","activation":"gelu","architectures":["DistilBertModel"],"attention_dropout":0.1,"dim":768,"dropout":0.1,"hidden_dim":3072,"initializer_range":0.02,"max_position_embeddings":512,"model_type":"distilbert","n_heads":12,"n_layers":6,"pad_token_id":0,"qa_dropout":0.1,"seq_classif_dropout":0.2,"sinusoidal_pos_embds":false,"tie_weights_":true,"transformers_version":"4.7.0","vocab_size":30522}"""
+  },
+  "created_time": 1694482261832,
+  "last_updated_time": 1694482324282,
+  "last_registered_time": 1694482270216,
+  "last_deployed_time": 1694482324282,
+  "total_chunks": 27,
+  "planning_worker_node_count": 1,
+  "current_worker_node_count": 1,
+  "planning_worker_nodes": [
+    "4p6FVOmJRtu3wehDD74hzQ"
+  ],
+  "deploy_to_all_nodes": true
+}
+```
+
+The response contains the model information. You can see that the `model_state` is `REGISTERED`. Additionally, the model was split into 27 chunks, as shown in the `total_chunks` field.
+</details>
+
+#### Advanced: Registering a custom model
+
+To register a custom model, you must provide a model configuration in the register request. For example, the following is a register request containing the full format for the model used in this tutorial:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
```

Comment:
I don't see `url` parameter in this request body, can you verify if this can work?

Reply:
This is one of the pretrained models. It worked when I ran the request.